app/scripts/directives/mobileIntegration.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileIntegration', {
+    controller: [
+      '$filter',
+      '$scope',
+      'APIService',
+      'Catalog',
+      'DataService',
+      MobileIntegration
+    ],
+    bindings: {
+      integrationServiceClass: '<',
+      onCreate: '<',
+      onDelete: '<',
+      consumerInstance: '<',
+      parameterData: '<'
+    },
+    templateUrl: 'views/directives/mobile-integration.html'
+  });
+
+  function MobileIntegration(
+    $filter,
+    $scope,
+    APIService,
+    Catalog,
+    DataService
+  ) {
+    var ctrl = this;
+    var instancePreferredVersion = APIService.getPreferredVersion('serviceinstances');
+    var bindingPreferredVersion = APIService.getPreferredVersion('servicebindings');
+    var isBindingReady = $filter('isBindingReady');
+    var watches = [];
+
+    ctrl.$onChanges = function(changes) {
+      if (changes.consumerInstance && changes.consumerInstance.currentValue &&
+          changes.integrationServiceClass && changes.integrationServiceClass.currentValue &&
+          !ctrl.watchesSet) {
+        var context = {namespace: _.get(ctrl, 'consumerInstance.metadata.namespace')};
+
+        DataService.list(instancePreferredVersion, context, function(serviceInstancesData) {
+          var data = serviceInstancesData.by('metadata.name');
+          ctrl.checkIntegrationInstance(data);
+
+          watches.push(DataService.watch(bindingPreferredVersion, context, function(bindingData) {
```

Comment:
As this happen onchange are we potentially adding more and more watches here? Not sure if we are just want to ask the question.

Reply:
The `ctrl.watchesSet` flag is used to only set the watches once.